src/azure/AzureTemplateFactory.cs

```diff
@@ -0,0 +1,43 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+//
+
+using AutoRest.Core;
+using AutoRest.TypeScript.azure.Templates;
+using AutoRest.TypeScript.Azure.Model;
+using AutoRest.TypeScript.Model;
+
+namespace AutoRest.TypeScript
+{
+    public class AzureTemplateFactory : TemplateFactory<CodeModelTSa>
+    {
+        public override Template<CodeModelTSa> CreateServiceClientTemplate(CodeModelTSa codeModel)
+        {
+            return new AzureServiceClientTemplate { Model = codeModel };
+        }
+        public override Template<CodeModelTSa> CreateServiceClientContextTemplate(CodeModelTSa codeModel)
```

Comment:
[nit] Missing empty line

Reply:
Fixed in https://github.com/Azure/autorest.typescript/pull/286/commits/379c8681252a6d3fae25fabe1162717c75390687.